src/NuGet.Clients/NuGet.PackageManagement.UI/DeepLinkURIParser.cs

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Versioning;
+
+namespace NuGet.PackageManagement.UI
+{
+    public static class DeepLinkURIParser
+    {
+        public static NuGetPackageDetails GetNuGetPackageDetails(string packageLink)
```

Comment:
You might consider a public TryParse method here instead that returns a bool and has an out parameter for the NuGetPackageDetails. You can still have the private TryParse method. But that would give more clarity over the result of the parsing rather than having the consumer check for null. Adjust tests accordingly.